OpenRA.Mods.Common/Traits/Render/WithAircraftLandingEffect.cs

```diff
@@ -0,0 +1,72 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Effects;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	[Desc("Plays an animation on the ground position when the actor lands.")]
+	public class WithAircraftLandingEffectInfo : ConditionalTraitInfo
+	{
+		[FieldLoader.Require]
+		public readonly string Image = null;
+
+		[SequenceReference(nameof(Image))]
+		public readonly string[] Sequences = { "idle" };
+
+		[PaletteReference]
+		public readonly string Palette = "effect";
+
+		[Desc("Should the sprite effect be visible through fog.")]
+		public readonly bool VisibleThroughFog = false;
+
+		[Desc("Time in ticks to play the animation after descending.")]
+		public readonly int LandingDelay = 0;
```

Comment:
I don't think this is the right approach. If you begin descending and cancel, the animation still plays. Perhaps you should set the `DistanceFromGround` here instead?

Reply:
Yes, that is also a lot easier to tweak. However, I now need to tick to calculate the distance, which is more expensive than using the landing interface alone.